streaming/src/main/scala/org/apache/spark/streaming/Checkpoint.scala

```diff
@@ -219,7 +223,9 @@ class CheckpointWriter(
           if (allCheckpointFiles.size > 10) {
             allCheckpointFiles.take(allCheckpointFiles.size - 10).foreach(file => {
               logInfo("Deleting " + file)
-              fs.delete(file, true)
+              if (fs.exists(file)) {
```

Comment:
I am not if this is necessary. These files that are being deleted came from a recent listing, so they are very very likely to exist.